test/api_getProduct_test.dart

```diff
@@ -862,5 +862,184 @@ void main() {
+        ProductField.CATEGORIES_TAGS_TRANSLATED,
+        ProductField.COUNTRIES_TAGS,
+        ProductField.COUNTRIES_TAGS_TRANSLATED,
+      ];
+
+      ProductQueryConfiguration englishConf = ProductQueryConfiguration(barcode,
+          language: OpenFoodFactsLanguage.ENGLISH, fields: fields);
+      ProductQueryConfiguration russianConf = ProductQueryConfiguration(barcode,
+          language: OpenFoodFactsLanguage.RUSSIAN, fields: fields);
+
+      // English!
+
+      ProductResult englishResult = await OpenFoodAPIClient.getProduct(
+          englishConf,
+          user: TestConstants.TEST_USER,
+          queryType: QueryType.TEST);
+      Product englishProduct = englishResult.product!;
+
+      expect(englishProduct.productName, equals('Pancakes'));
+      expect(englishProduct.productNameTranslated, equals('Pancakes'));
+
+      expect(englishProduct.ingredientsText, equals('Flour, water'));
+      expect(englishProduct.ingredientsTextTranslated, equals('Flour, water'));
+
+      expect(englishProduct.ingredientsTags, equals(['en:flour', 'en:water']));
+      expect(
+          englishProduct.ingredientsTagsTranslated, equals(['Flour', 'Water']));
+
+      expect(englishProduct.categoriesTags, equals(['en:beverages']));
+      expect(englishProduct.categoriesTagsTranslated, equals(['Beverages']));
+
+      expect(englishProduct.countriesTags, equals(['en:russia']));
+      expect(englishProduct.countriesTagsTranslated, equals(['Russia']));
+
+      // Russian!
+
+      ProductResult russianResult = await OpenFoodAPIClient.getProduct(
+          russianConf,
+          user: TestConstants.TEST_USER,
+          queryType: QueryType.TEST);
+      Product russianProduct = russianResult.product!;
+
+      expect(russianProduct.productName, equals('Pancakes'));
+      expect(russianProduct.productNameTranslated, isNull);
+
+      expect(russianProduct.ingredientsText, equals('Flour, water'));
+      expect(russianProduct.ingredientsTextTranslated, isNull);
+
+      expect(russianProduct.ingredientsTags, equals(['en:flour', 'en:water']));
+      expect(
+          russianProduct.ingredientsTagsTranslated, equals(['Мука', 'Вода']));
+
+      expect(russianProduct.categoriesTags, equals(['en:beverages']));
+      expect(russianProduct.categoriesTagsTranslated, equals(['Напитки']));
+
+      expect(russianProduct.countriesTags, equals(['en:russia']));
+      expect(russianProduct.countriesTagsTranslated, equals(['Россия']));
+    });
+
+    test('translated fields when product is translated into a second language',
+        () async {
+      String barcode = '2222222222222';
+
+      Product englishInputProduct = Product(
+        barcode: barcode,
+        lang: OpenFoodFactsLanguage.ENGLISH,
+        productNameTranslated: 'Pancakes',
+        ingredientsTextTranslated: 'Flour, water',
+        categories: 'Beverages',
+        countries: 'Russia',
+      );
+
+      Product russianInputProduct = Product(
```

Comment:
Looking at the test code, I'm thinking it would make sense to add a "translatedLang" field directly in the Product class, that would indicate what is the language of all the *Translated fields.

So when we request products with translated fields, we save the corresponding language to translatedLang.

And when we save the product, instead of passing lc to saveProduct, it justs uses the value of translatedLang.

What do you think?

Reply:
Can do!